### PR TITLE
Fixed support for null and false values when binding parameters

### PR DESCRIPTION
**Pull request by alanhartless**
## Summary of the changes / Why this is an improvement

When using Doctrine ORM, two cases failed due to the PDOStatement class type casting the values incorrectly:

1. null values - null values were type cast to the underlying data type (int, bool, timestamp). This means that setting null parameters resulted in at best `0` for bool, `0` for int, `''` for string, etc  or at worst, a `cannot cast '' to type timestamp` SQL error. This is fixed by checking if the value is null and if so, returning null without typecasting.
2. false values - Doctrine passes a value of "false" (a string) with the type as PDO::PARAM_BOOL. Typecasting with `(bool) "false"` results in `true` making it impossible to set a false value unless an empty string or 0 is passed. This fixes that by using PHP's `filter_var` with `FILTER_VALIDATE_BOOLEAN` so that `"false" = false`, `"true" = true`, `1 = true`, `0 = false` and so forth. 

## Checklist

 - [ x ] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by alanhartless** [comment]
@seut You're welcome! I'd be glad to. Where do I write the user-facing changes? The PR description, commit it to https://github.com/crate/crate-pdo/blob/master/CHANGES.rst#unreleased or somewhere else? Thanks!

**Comment by seut** [comment]
@alanhartless Yes that's the right place.

**Comment by alanhartless** [comment]
@seut Is https://github.com/crate/crate-pdo/pull/94/commits/f1977b434220bc67e51d5e9119b4db8adc9247b4 OK?

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,9 @@
 Unreleased
 ==========
 
+- Fix: binding NULL as a parameter would not persist as NULL or cause an error depending on the column data type
```
Comment (seut) [inline CHANGES.rst]:
```suggestion
- Fixed `cast` exception when binding a NULL value to a parameter which is not of type PDO::PARAM_NULL.
```

File: CHANGES.rst [hunk 2]
```diff
@@ -4,6 +4,9 @@
...
+- Fix: binding FALSE as a parameter would persist as TRUE
```
Comment (seut) [inline CHANGES.rst]:
```suggestion
- Fixed boolean parsing when binding a PDO::PARAM_BOOL parameter value.
```

**Comment by seut** [comment]
@alanhartless Thanks a lot. I'll squash (if you don't mind) and merge your commits asap. Will notify you once released.

**Comment by alanhartless** [comment]
Don't mind a bit. Thanks!

**Comment by seut** [comment]
@alanhartless We've just pushed a new 1.0.1 release including these fixes. Thanks again for contributing.

**Comment by alanhartless** [comment]
Awesome! Thank you!

**Pull request merged**